### DISCUSSION TITLE
Ship Bun w/ Node.js aliasing only production-ready features

### DISCUSSION THREAD

**Discussion by o-alexandrov** [oven-sh/bun#1846] [Ideas]
To speed up adoption, how about shipping Bun and Node.js together?
- this way, you could alias only production-ready features from Bun when accessing Node.js built-in modules

1. Wouldn't it have any performance benefits over use of only Node.js modules?
2. Is there anything preventing to do so?

P.S. I'm new to Bun, so if I'm fundamentally wrong, please still express why

**Comment by vjpr** [comment]
Interesting idea but I think too complicated in practice.

Integrating a Node.js build would be too much work probably and suck time away from just implementing features in Bun.

Alternatively you could do IPC calls for missing things to a sidecar Node process, but that would be too slow.

Would be nice to have some kind of "launcher" tool that would try run the script with Bun first, and then check the error messages and see if they are related to lack of compatibility, and then just run in Node. So then `node` could be replaced with `bun` everywhere, or simply `node` aliased to `bun`. Could also do quick check about which version is being used and diagnose those kinds of errors.
